### PR TITLE
Create deps.yaml

### PR DESCRIPTION
**Pull request by bms63**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/dev/CONTRIBUTING.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: .github/workflows/deps.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Dependency Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```
[the remark this answers is not in the document]
Comment (bms63) [reply .github/workflows/deps.yaml:9]: Oh I was just testing it out to see if it found our bug.

hadn’t really got further then that! :)

**Comment by bms63** [comment]
> Are you sure you want to keep this in admiraldev?

No I think it should live in admiralci so folks can grab if needed.

@bundfussr and @ddsjoberg this looks interesting https://github.com/insightsengineering/teal.code/blob/main/.github/workflows/scheduled.yaml

It is sort of what I implemented here.

Reading through the log file I see 
image

@cicdguy is it using 4.4 or 4.0?  Is there a setting I need to toggle for it to use the minimum version recommended in the DESCRIPTION file.

Thanks for your help too!!!!!!!!!

**Comment by bundfussr** [comment]
> @bundfussr and @ddsjoberg this looks interesting https://github.com/insightsengineering/teal.code/blob/main/.github/workflows/scheduled.yaml

@bms63 , there is also https://github.com/insightsengineering/r-verdepcheck-action/tree/main which is referenced in the `verdepcheck` documentation. I'm not sure what the difference is and which one is better.

**Comment by bms63** [comment]
> > @bundfussr and @ddsjoberg this looks interesting https://github.com/insightsengineering/teal.code/blob/main/.github/workflows/scheduled.yaml
> 
> @bms63 , there is also https://github.com/insightsengineering/r-verdepcheck-action/tree/main which is referenced in the `verdepcheck` documentation. I'm not sure what the difference is and which one is better.

I think the schedule one might be better - run it on main every month rather than in every PR.  as suggested by @cicdguy

**Comment by bms63** [comment]
> > > > @bundfussr and @ddsjoberg this looks interesting https://github.com/insightsengineering/teal.code/blob/main/.github/workflows/scheduled.yaml
> > > 
> > > 
> > > @bms63 , there is also https://github.com/insightsengineering/r-verdepcheck-action/tree/main which is referenced in the `verdepcheck` documentation. I'm not sure what the difference is and which one is better.
> > 
> > 
> > I think the schedule one might be better - run it on main every month rather than in every PR. as suggested by @cicdguy
> 
> I'm not sure if we need a scheduled run. I would run it before we submit to CRAN, i.e., if `[skip vbump]` is in the title of the PR. Additionally there should be the option to execute the workflow manually. Then you could run it when you use a function that you are not sure is available in the minimum version.

ooooh i like that `[skip vbump]`.  I can put it in my Release notes to run manually as we get close to a release.  https://github.com/pharmaverse/admiral/discussions/2394

**Comment by bms63** [comment]
See https://github.com/pharmaverse/admiral/pull/2489

**Pull request closed without merging**